core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeSorterSpillReader.java

```diff
@@ -69,8 +70,8 @@ public UnsafeSorterSpillReader(
       bufferSizeBytes = DEFAULT_BUFFER_SIZE_BYTES;
     }
 
-    final BufferedInputStream bs =
-        new BufferedInputStream(new FileInputStream(file), (int) bufferSizeBytes);
+    final NioBasedBufferedFileInputStream bs =
```

Comment:
BTW I think IndexShuffleBlockResolver, RowQueue have instances that could use a similar treatment.

Can this reference be `InputStream`?